packages/typespec-azure-portal-core/src/decorators.ts

```diff
@@ -0,0 +1,253 @@
+    .filter((value: Type) => value.kind === "String")
+    .map((value: Type) => (<StringLiteral>value).value);
+}
+
+export function getAboutLearnMoreDocs(program: Program, target: Type) {
+  const about = getAbout(program, target);
+  return about.properties.get("learnMoreDocs");
+}
+
+export function getAboutLearnMoreDocsItemsArray(program: Program, target: Type) {
+  const learnMoreDocs = getAboutLearnMoreDocs(program, target);
+  return learnMoreDocs.type.values
+    .filter((value: Type) => value.kind === "String")
+    .map((value: Type) => (<StringLiteral>value).value);
+}
+
+export function $marketplaceOffer(
+  context: DecoratorContext,
+  target: Model,
+  options: marketplaceOfferOptions
+) {
+  const { program } = context;
+  validateDecoratorUniqueOnNode(context, target, $marketplaceOffer);
+  isARMResource(program, target, "marketplaceOffer");
+  if (options && (options as Model).properties) {
+    const id = (options as Model).properties.get("id");
+    if (id?.type.kind === "String") {
+      if (id.type.value.match(/\s/)) {
+        reportDiagnostic(program, {
+          code: "invalidType",
+          messageId: "marketplaceOfferId",
+          target,
+        });
+      }
+    }
+  }
+  program.stateMap(PortalCoreKeys.marketplaceOffer).set(target, options);
+}
+
+export function getMarketplaceOfferId(program: Program, target: Type) {
+  const marketplaceOffer = program.stateMap(PortalCoreKeys.marketplaceOffer).get(target);
+  return marketplaceOffer.properties.get("id");
+}
+
+// export function $patternValidationMessage(context:DecoratorContext, target: Scalar | ModelProperty, message: string) {
+//   const { program } = context;
+//   validateDecoratorUniqueOnNode(context, target, $patternValidationMessage);
+//   validateDecoratorNotOnType(context, target, $pattern, $patternValidationMessage);
+//   const patternSymbol = Symbol.for("TypeSpec.pattern");
+//   const pattern = program.stateMap(patternSymbol).get(target);
+//   const pat = getPattern(program, target);
+//   program.stateMap(PortalCoreKeys.patternValidationMessage).set(target, message);
+// }
+
+// export function getPatternValidationMessage(program: Program, target: Type): string | undefined {
+//   const result = program.stateMap(PortalCoreKeys.patternValidationMessage).get(target);
+//   return result && result.value;
+// }
+
+export function $displayName(context: DecoratorContext, target: ModelProperty, name: string) {
+  const { program } = context;
+  validateDecoratorUniqueOnNode(context, target, $displayName);
+  program.stateMap(PortalCoreKeys.displayName).set(target, name);
+}
+
+export function getDisplayName(program: Program, target: Type): string | undefined {
+  return program.stateMap(PortalCoreKeys.displayName).get(target);
+}
+
+export function getDisplayNameValue(program: Program, target: Type): string | undefined {
+  const result = program.stateMap(PortalCoreKeys.displayName).get(target);
+  return result && result.value;
+}
```

Comment:
Something a little weird is going on here. The first function doesn't actually return `string`, but a StringLiteral object from the compiler. I would suggest just making `getDisplayName` do what `getDisplayNameValue` does and only having one of them.